### PR TITLE
Added  DEFENDER_DISABLE_IP_LOCKOUT so that you can disable IP lockouts if you want

### PR DESCRIPTION
**Pull request by kencochrane**
I added the ability to disable the IP locking features, when enabled it will only lock out based on username. 

This would be useful if you have a lot of users that connect via the same proxy, and you want to prevent everyone from getting locked out because of one user's bad login attempts.

I also updated the travisCI settings, to test for newer versions of Django and python 3.5

bumped version to 0.4.0

/cc @bcwalrus @marcusmartins

### CODE REVIEW

File: defender/utils.py [hunk 1]
```diff
@@ -243,15 +267,24 @@ def lockout_response(request):
...
+        ip_blocked = False
 
     # username blocked?
+    username = request.POST.get(config.USERNAME_FORM_FIELD, None)
     user_blocked = REDIS_SERVER.get(get_username_blocked_cache_key(username))
 
+    if config.DISABLE_IP_LOCKOUT:
+        # if DISABLE_IP_LOCKOUT is turned on, then return user_blocked
+        # result, since we don't care about ip at this point.
+        return user_blocked
```
Comment (bcwalrus) [inline defender/utils.py]: I don't think we need this condition. Instead, on line 295, get rid of the big `else` clause, and just do `return ip_blocked or user_blocked`.

Btw, line 290 should just be `return ip_blocked and user_blocked`. No need to do an `if`.

**Comment by bcwalrus** [comment]
Looks good. A small comment on cleaning up the boolean logic. Thanks @kencochrane!

**Comment by marcusmartins** [comment]
lgtm

**Comment by kencochrane** [comment]
@bcwalrus I made the change you suggested, if all unit tests pass, I'll cut a new build and push to pypi.

**Comment by bcwalrus** [comment]
LGTM. Thanks!

**Pull request merged**